reconcilers/validation.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright 2025 the original author or authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package reconcilers
+
+import "context"
+
+type Validator interface {
+	Validate(ctx context.Context) error
+}
+
+type nestedValidationKey struct{}
+
+func WithNestedValidation(ctx context.Context) context.Context {
```

Comment:
Use a [stasher](https://pkg.go.dev/reconciler.io/runtime@main/reconcilers#Stasher)

Reply:
@scothis  Great idea! However, I've moved validation into its own `validation` package. To break a cyclic dependency on `reconcilers`, I would have to move stash stuff into, say, `stash`. That's ok I think. But before I plow on, I'd like you to acknowledge.

Reply:
Sounds good, so long as we alias back the types to not break anyone.

Reply:
@scothis After starting the work I realised it's not possible without breaking anyone, b/c generic types cannot be easily aliased (yet). Recorded https://github.com/reconcilerio/runtime/issues/591 instead